inst/resources/js/config.js

```diff
@@ -46,6 +46,56 @@
     if (beforePaged) await beforePaged();
   };
 
+  // from https://stackoverflow.com/q/21647928
+  const toUTF16BE = x => {
+    let res = '';
+    for (i=0; i < x.length; i++) {
+      let hex = x.charCodeAt(i).toString(16);
+      hex = ('000' + hex).slice(-4);
+      res += hex
+    }
+    res = 'feff' + res ;
+    return res;
+  }
+
+  const findPage = el => {
+    while (el.parentElement) {
+      el = el.parentElement;
+      if (el.getAttribute('data-page-number')) {
+        return parseInt(el.getAttribute('data-page-number'));
+      }
+    }
+    return null;
+  };
+
+  const tocEntriesInfos = ul => {
+    const tocEntries = ul.children; // tocEntries are 'li' elements
+    let result = []; // where we store the results
```

Comment:
Sorry, I missed to implement a control to check whether there is a TOC in the document. These lines will fail if there is no TOC.

I think we could replace these lines with:

```suggestion
    let result = []; // where we store the results
    // if there is no element, return an empty array 
    if (!ul) {
      return result;
    }
    const tocEntries = ul.children; // tocEntries are 'li' elements
```